File: react/src/interfaces/restaurant.ts
import { UUID } from "crypto";
import { Availability } from "./availability";
import { Review } from "./review";

export interface Restaurant {
  id: UUID;
  name: string;
  cuisine?: string;
  rating?: number | string;
  totalReviews?: number;
  location: string;
  phone: string;
  email: string;
  image: string; // geralmente `photos[0]`
  description: string;
  priceRange: string;// vindo de availability
  availableHours: string[];     // idem
  availableTables: any;
  reviews: any;
  availabilitys: Availability[]
  hours: any,
}

File: api/src/main/java/com/reservafacil/api/entities/Restaurant.java
package com.reservafacil.api.entities;

import jakarta.persistence.*;
import lombok.Data;
import lombok.Getter;
import lombok.Setter;

import java.util.ArrayList;
import java.util.List;
import java.util.UUID;

@Entity
@Getter
@Setter
public class Restaurant {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;
    private String name;
    private int nif;
    private String address;
    private String coordinates;
    private String email;
    private String phone;
    private String price_range;
    private String description;
    private String photos;
    @OneToMany(mappedBy = "restaurant", fetch = FetchType.EAGER)
    private List<Tables> tables = new ArrayList<>();
    @OneToMany(mappedBy = "restaurant")
    private List<Employee> employees = new ArrayList<>();
    @OneToMany(mappedBy = "restaurant")
    private List<Review> reviews = new ArrayList<>();
    @OneToMany(mappedBy = "restaurant", fetch = FetchType.EAGER)
    private List<Availability> availabilities = new ArrayList<>();
}
